### PR TITLE
Fix: Add initial readme

### PR DESCRIPTION
**Pull request by Tombre**
Note that this is a "fix" so the npm package will be published, thus updating that version of documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,4 +2,196 @@
...
+        six: arrayOf('number'),
+        seven: 'unknown',
+        eight: optional('boolean')
+    }
+)
+```
+
+### Options
+
+-   **coerceValidObjectIntoArray** (boolean): If you are validating an array with `arrayOf`, setting this to `true` will convert those properties to arrays for you, rather than returning an error.
```
Comment (JakeGinnivan) [inline README.md]: Could always put a comment that this is useful when using xml to json libraries like ....

**Review by JakeGinnivan — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Tombre) [reply README.md]: Good idea, added a note.

**Review by cwoodley — APPROVED** [review]

**Comment by JakeGinnivan** [comment]
Could also mention under a similar projects section https://github.com/gcanti/io-ts

The main reason we didn't go with io-ts was the complexity and the lack of coersion features. We primarily wanted a way to normalise and validate a large parsed xml structure.

File: README.md [hunk 2]
```diff
@@ -2,4 +2,196 @@
 
 [![Build Status](https://travis-ci.com/sevenwestmedia-labs/typescript-object-validator.svg?branch=master)](https://travis-ci.com/sevenwestmedia-labs/typescript-object-validator)
 
-A simple library for typescript projects for validating object shapes.
+A simple library for typescript projects to validating object shapes. You can use this package to declare the shape you'd like an object to align to. This is handy when you have a complex object (for example, an API response) and want to validate the object, AND have typescript recognize the shape.
```
Comment (luke-john) [inline README.md:5]: The first sentence here has a typo/odd word usage.
```diff
- A simple library for typescript projects to validating object shapes.
+ A simple library for typescript projects for validating object shapes. 
```
(the double for in the diff suggestion reads a bit weird, so you might want to do something different 😉 )

File: README.md [hunk 3]
```diff
@@ -2,4 +2,196 @@
...
+    validationItem,
+    expectedObjectShape,
+    validationOptions
+)
+```
+
+-   **objectDescription** (required - string): A description which is used in error messages when the object does not validate
+-   **validationItem** (required - object): An object you want to validate
+-   **expectedObjectShape** (required - object): A definition of what shape the object should match
+-   **validationOptions** (required - object): A set of options to change the validation behaviour, [see more](#Options)
```
Comment (luke-john) [inline README.md:75]: Marked a required here, but not included in the Basic Usage example.

File: README.md [hunk 4]
```diff
@@ -2,4 +2,196 @@
...
+    {
+        age: '30',
+        meta: { group: 'Staff' }
+        names: [
+            { fname: 'Jeff', lname: 'Thompson' },
+            { fname: 'Jeff', lname: 'Thompson' }
+        ]
+    },
+    {
+        age: number,
```
Comment (luke-john) [inline README.md:131]: looks like this is missing quotes.
```diff
-        age: number,
+        age: 'number',
```

**Review by luke-john — COMMENTED** [review]
Looks really cool, would have been super handy in a bunch of api handling I've done in the past.

Nice work

**Comment by JakeGinnivan** [comment]
Nice @luke-john, either @Tombre or I will put up a PR tomorrow sorting these issues out. nice review

**Pull request merged**